code/modules/atmospherics/machinery/components/binary_devices/dp_vent_pump.dm

```diff
@@ -51,6 +51,11 @@
 /obj/machinery/atmospherics/components/binary/dp_vent_pump/process_atmos()
 	if(!on)
 		return
+
+	var/turf/location = get_turf(loc)
+	if(isclosedturf(location))
+		return PROCESS_KILL
```

Comment:
Don't do this. We don't manage object processing like this. just have it return nothing

Reply:
okok

Reply:
Plenty of things handle their processing like this and disabling/enabling your processing based on events is generally a better way to go about things

Reply:
yeah, but vents don't right now, so this would just end up making them stop forever